src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -531,6 +532,23 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithTemplateAnd
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises : Queries a programming exercise by its project key.
+     *
+     * @return the ProgrammingExercise of this project key in an ResponseEntity or 404 Not Found if no exercise exists
+     */
+    @GetMapping("programming-exercises/project-key/{projectKey}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<ProgrammingExercise> getExerciseByProjectKey(@PathVariable String projectKey) {
+        try {
+            final ProgrammingExercise exercise = programmingExerciseRepository.findOneByProjectKeyOrThrow(projectKey, false);
+            return ResponseEntity.ok(exercise);
```

Comment:
If  you use `@EnforceAtLeastStudent` instead of `@EnforceAtLeastStudentInExercise` you need to add an explicit check here. 
```suggestion
            authCheckService.checkIsAtLeastRoleInExerciseElseThrow(Role.Student, null);
            return ResponseEntity.ok(exercise);
```